cartographer/io/mapping_state_deserializer.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_IO_MAPPING_STATE_DESERIALIZER_H_
+#define CARTOGRAPHER_IO_MAPPING_STATE_DESERIALIZER_H_
+
+#include <iterator>
+
+#include "cartographer/io/proto_stream_interface.h"
+#include "cartographer/mapping/proto/pose_graph.pb.h"
+#include "cartographer/mapping/proto/serialization.pb.h"
+#include "cartographer/mapping/proto/trajectory_builder_options.pb.h"
+
+namespace cartographer {
+namespace io {
+
+// Class to help deserializing a previously serialized mapping state from a
+// proto stream, abstracting away the format parsing logic.
+class MappingStateDeserializer {
```

Comment:
Will all deserialization go through this later? Or will it be used by the ground truth and ros tools that only need the pose graph?
Rename. Perhaps PoseGraph/ProtoStreamDeserializer?

Reply:
Renamed.